pkg/semtech-loramac/contrib/semtech_loramac_radio.c

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright (C) 2017 Fundacion Inria Chile
+ *               2017 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ * @ingroup     pkg_semtech-loramac
+ * @file
+ * @brief       Compatibility functions for controlling the radio driver
+ *
+ * @note        Even if the function names are related to SX1276, they use
+ *              the RIOT SX127x driver and can be used with SX1272 and SX1276
+ *              devices.
```

Comment:
I agree with that but what if someone want to reuse some old code using SX1272 ?
That person would have to rename everything ?

Reply:
> what if someone want to reuse some old code using SX1272 ?

What do mean with "some old code using sx1272" ? In this PR using sx1276 or sx1272 is just a matter of using the corresponding driver module in the application `Makefile`, e.g `USEMODULE += sx1272` or `USEMODULE += sx1276`.
The upstream LoRaMAC code uses its own HAL and in the package we bypass it to reuse RIOT's internal driver. The RIOT driver avoids the code duplication that is done in LoRaMAC code.

Reply:
Sorry If I was unclear, what I mean is the following : 
Let's say someone what to reuse an old code using SX1272.
With the current state, functions like SX1272Init  cannot be used. 

I'm not saying it's a no-go (and I have no better proposition right now) but we should at least make it clear.

Reply:
ok, I see. This use case is not covered unfortunately, and I think it won't be. In this PR, the proposal is to go through a custom RIOT API (see semtech-loramac.h) that provide basic functions for getting/setting mac parameters, joining a network and sending/receiving packets. This is more or less what is provided by LoRaMAC applications (see their main.c's)

Reply:
I guess it is fine for now. We should just write it somewhere to warn the current limitation of this pkg. What do you think ?

Reply:
Personnally, I don't see it as a limitation but rather as an improvement regarding the original API provided by Semtech. Have you looked at their application code ? I just wrapped it in semtech-loramac.c (except the link check and compliance test) and added an API for mainpulating the MAC on top of their internal code. The MAC is isolated in a thread so that all calls to the MAC timers and other stuff are not processed in the ISR context (which is what they do basically) but in a separate thread context.
I think IMHO that in term of user experience, this is not comparable.

Reply:
> Have you looked at their application code

Unfortunately yes... I don't like it at all.

I guess the word 'limitation' was not a suitable one. 
People who will use this pkg  probability  have already used the original semtech loramac. This is why we must inform them (wiki page ?) the current differences (not limitations) from the original loramac.

Reply:
> This is why we must inform them (wiki page ?) the current differences (not limitations) from the original loramac.

The simplest is to add note in the doc.txt of the package. This will be in the doxygen documentation.

Reply:
I just pushed a detailed doxygen documentation with a warning.

Reply:
Thanks @aabadie !

Reply:
ping @aabadie